.github/workflows/unit.yml

```diff
@@ -41,25 +41,37 @@ jobs:
         id: global-cache-dir-path
         run: |
           echo "yarn=$(yarn cache dir)" >> $GITHUB_OUTPUT
-          echo "terraform=$(mktemp -d)" >> $GITHUB_OUTPUT
-          echo "go=$(mktemp -d)" >> $GITHUB_OUTPUT
-          echo "providerSchema=$(mktemp -d)" >> $GITHUB_OUTPUT
+          mkdir -p /usr/local/share/.cache/terraform
+          echo "terraform=/usr/local/share/.cache/terraform" >> $GITHUB_OUTPUT
+          mkdir -p /usr/local/share/.cache/go
+          echo "go=/usr/local/share/.cache/go" >> $GITHUB_OUTPUT
+          mkdir -p /usr/local/share/.cache/providerSchema
+          echo "providerSchema=/usr/local/share/.cache/providerSchema" >> $GITHUB_OUTPUT
       - uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8
         with:
           path: ${{ steps.global-cache-dir-path.outputs.yarn }}
-          key: yarn-${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
+          key: yarn-${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}-unit-${{ inputs.package }}
+          restore-keys: |
+            yarn-${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}-unit-
+            yarn-${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}-
+            yarn-${{ runner.os }}-
       - uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8
         with:
           path: ${{ steps.global-cache-dir-path.outputs.terraform }}
-          key: terraform-${{ runner.os }}-${{ inputs.terraform_version }}
+          key: terraform-${{ runner.os }}-${{ inputs.terraform_version }}-unit-${{ inputs.package }}
+          # restore-keys: |
```

Comment:
Stray comment

Reply:
Maybe accidental comment now that I read it?

Reply:
It was intended and did fix the unit tests. I should remove it entirely instead of just commenting it out though (as the fix worked).

Reply:
replaced it with a comment explaining why we don't have restore-keys for unit tests